src/main/java/com/salesforce/mirus/MirusSourceTask.java

```diff
@@ -188,7 +191,18 @@ private void seekToOffsets(List<TopicPartition> partitionIds) {
 
   List<SourceRecord> sourceRecords(ConsumerRecords<byte[], byte[]> pollResult) {
     List<SourceRecord> sourceRecords = new ArrayList<>(pollResult.count());
-    pollResult.forEach(sourceRecord -> sourceRecords.add(toSourceRecord(sourceRecord)));
+    pollResult.forEach(
+        consumerRecord -> {
+          SourceRecord sourceRecord = toSourceRecord(consumerRecord);
+          Long sourceOffset = consumerRecord.offset();
+          Long latestOffset = latestOffsetMap.get(sourceRecord.sourcePartition());
+
+          // Skip any record that has already been handled by this task
+          if (latestOffset == null || sourceOffset > latestOffset ) {
```

Comment:
This will cause some data loss from the reset:

1. Consumers consumes to offset 100 from old leader
1. Old leader shuts down and new leader starts up at offset 95
1. Mirus resets offset to zero and consumes to 95, skipping messages as expected
1. Mirus consumes messages 96-100 which are net-new (produced to new leader after leadership transfer), _but skips them, causing data loss_
1. Mirus consumes messages 101+ and replicates normally.

Reply:
Now using a configurable overlap window to cover cases where the risk of dropping some records is not acceptable but  few duplicates are OK(few compared with a full partition replay, that is).